File: backend/app/models/assessment.rb
class Assessment < ApplicationRecord
  belongs_to :client
  belongs_to :staff
  belongs_to :support_plan, optional: true

  validates :assessment_type, presence: true
  validates :assessment_date, presence: true
  validates :summary, presence: true, if: -> { status == 'approved' }
  validates :overall_score, numericality: { greater_than_or_equal_to: 1, less_than_or_equal_to: 100 }, allow_nil: true
  validates :status, inclusion: { in: %w[draft pending approved] }

  scope :by_date, -> { order(assessment_date: :desc) }
  scope :recent, -> { where(assessment_date: 1.month.ago..Date.today) }
  scope :approved, -> { where(status: 'approved') }

  ASSESSMENT_TYPES = %w[initial periodic annual discharge].freeze

  def finalize!(staff)
    update!(
      status: 'approved',
      finalized_at: Time.current,
      finalized_by: staff.id
    )
  end

  def editable?
    status == 'draft'
  end
end

File: frontend/src/types/assessment.ts
export interface Assessment {
  id: number;
  clientId: number;
  staffId: number;
  supportPlanId?: number;
  assessmentType: 'initial' | 'periodic' | 'annual' | 'discharge';
  assessmentDate: string;
  summary?: string;
  overallScore?: number;
  categoryScores?: Record<string, number>;
  strengths?: string;
  challenges?: string;
  recommendations?: string;
  goals?: string;
  status: 'draft' | 'pending' | 'approved';
  finalizedAt?: string;
  finalizedBy?: number;
  attachments?: string;
  notes?: string;
  createdAt: string;
  updatedAt: string;
}

export interface AssessmentFormData {
  clientId: number;
  staffId: number;
  supportPlanId?: number;
  assessmentType: 'initial' | 'periodic' | 'annual' | 'discharge';
  assessmentDate: string;
  summary?: string;
  overallScore?: number;
  categoryScores?: Record<string, number>;
  strengths?: string;
  challenges?: string;
  recommendations?: string;
  goals?: string;
  status?: 'draft' | 'pending' | 'approved';
  notes?: string;
}

export const ASSESSMENT_TYPES = {
  initial: '初回',
  periodic: '定期',
  annual: '年次',
  discharge: '終了時',
} as const;

export const ASSESSMENT_STATUS = {
  draft: '下書き',
  pending: '承認待ち',
  approved: '承認済み',
} as const;

export const ASSESSMENT_CATEGORIES = [
  '身体機能',
  '認知機能',
  '社会参加',
  '日常生活動作',
  'コミュニケーション',
] as const;